neuralnetwork/machinelearning/NeuralNetwork.java

```diff
@@ -1,76 +1,87 @@
 package neuralnetwork.machinelearning;
 
 import neuralnetwork.activatefunction.ActivateFunction;
-import neuralnetwork.inputfunction.InputFunction;
+import neuralnetwork.activatefunction.Softmax;
+import neuralnetwork.inputfunction.InputNormalize;
 import neuralnetwork.layer.HiddenLayer;
 import neuralnetwork.layer.InputLayer;
 import neuralnetwork.layer.LayerData;
 import neuralnetwork.layer.OutputLayer;
 
 import java.util.ArrayList;
 import java.util.Iterator;
+import java.util.List;
 import java.util.ListIterator;
 
+import static neuralnetwork.machinelearning.NetWorkConfig.*;
+
 public class NeuralNetwork {
-    private InputLayer inputLayer = null;
+    private InputLayer inputLayer;
     private int inputLayerSize;
-    private ArrayList<HiddenLayer> hiddenLayers = new ArrayList<>();
-    private ArrayList<Integer> hiddenLayersSize = new ArrayList<>();
-    private OutputLayer outputLayer = null;
-    private double learningRate;
-    private ArrayList<int[]> layerSize = new ArrayList<>();
-
-    public NeuralNetwork(double learningRate) {
+    private final ArrayList<HiddenLayer> hiddenLayers = new ArrayList<>();
+    private final ArrayList<Integer> hiddenLayersSize = new ArrayList<>();
+    private OutputLayer outputLayer;
+    private final double learningRate;
+    private final List<int[]> layerSize = new ArrayList<>();
+    private final InputNormalize inputNormalize = getInputNormalize();
+    private final ActivateFunction activateFunction = getFunction();
+    private final Softmax softmax = getSoftMax();
+
+    private NeuralNetwork(double learningRate) {
         this.learningRate = learningRate;
     }
 
-    public void addInputLayer(InputFunction inputFunction, int size) throws Exception {
+    public static NeuralNetwork rate(double learningRate) {
+        return new NeuralNetwork(learningRate);
+    }
+
+    public NeuralNetwork addInputLayer(int size) throws Exception {
```

Comment:
By this way, we can't set input function. Although there is only one input function now... It should have inputFunction as parameters